### DISCUSSION TITLE
Bulk Operation error - SyntaxError: "undefined" is not valid JSON

### DISCUSSION THREAD

**Discussion by Deanobeano2** [cortexrd/Knack-Toolkit-Library#419] [Q&A] [answered]
Hi

I'm attempting to use the Bulk Operation feature for the 1st time in quite a while and am encountering a pop-up error of  "SyntaxError: "undefined" is not valid JSON" when attempting to update a date field, in dd/mm/yyyy format, in several records.

 Is this possibly related to the date change recently made by Knack?

Any thoughts or suggestions would be appreciated, though this is something we can make do without in the meantime as no doubt most if not all of us are busy as we rapidly approach the festive/holiday season.

Dean

**Comment by cortexrd** [comment]
Hi Dean,

I've made a hotfix in v0.29.4 for this.  Are you using that version?

**Comment by Deanobeano2** [comment]
Yes

**Comment by cortexrd** [comment]
Can you give me access to that page?  I'm in your app now.

**Comment by Deanobeano2** [comment]
I believe you're still set up with Bulk Edit User Role.

**Comment by cortexrd** [comment]
I need the Bulk Edit role also.
For the page, just give me the view ID and I'll find the link.

**Comment by cortexrd** [comment]
No, only Manager

**Comment by Deanobeano2** [comment]
View_1508.

What I was trying to update:

![image](https://github.com/user-attachments/assets/2626f342-e2a3-42b1-a734-27c6c719ea4e)

**Comment by Deanobeano2** [comment]
Done.

**Comment by cortexrd** [comment]
From what I see, you didn't add me the Bulk Edit role.  Instead, you added this keyword to the view:

![image](https://github.com/user-attachments/assets/45f6a395-c238-4370-9ed2-2f3e4c21de06)

But that doesn't allow me to update anything, just copy.

**Comment by Deanobeano2** [comment]
Sorry, there are 2 users for you in the system and I've just updated the 2nd 1 so they are both Manager and Bulk Edit User Roles.

**Comment by cortexrd** [comment]
I can now reproduce the bug at will.  But it's going to be more complicated that I thought to fix it.
Yes, it's related to Knack's change, combined to my incomplete handling of all use cases.  Date/Time fields are sooo complicated!

I'll continue tomorrow, it's late now.

Thanks for your patience and for reporting this issue.

Norm

**Comment by Deanobeano2** [comment]
Hi Norm

 

I’m glad it was a bug and not me wasting your time again.

 

I know you’re busy so please don’t expend any time on this on our account.

 

It’s time for you to forget about this until another day and no doubt head off to bed.

 

Thank you once again for all of your help.

 

Kind regards

Dean Morrison

General Manager
Excell Cleaning Services Limited

**Comment by cortexrd** [comment]
Dean, can you test the latest release please?
**v0.29.5**
And update the loader:  https://github.com/cortexrd/Knack-Toolkit-Library/blob/dev/KTL_Loader.js

**Comment by Deanobeano2** [comment]
Hi Norm

It's back working as it used to, and I've updated the loader.

Many thanks, once again.

Dean

**Reply by cortexrd** [reply, marked as answer]
Don't forget to click **Mark as answer** in the preceding post!
I can do it, but I prefer when it's the original poster who does it :)

**Comment by Deanobeano2** [comment]
Done

 

Kind regards

Dean Morrison

General Manager
Excell Cleaning Services Limited
